### DISCUSSION TITLE
Process profiling and metrics

### DISCUSSION THREAD

**Discussion by Leokuma** [Hexagon/pup#17] [Ideas]
I'm planning to make a [Node Exporter](https://github.com/prometheus/node_exporter) plugin for Pup.

I'm also planning to make a [Pidusage](https://github.com/soyuka/pidusage) plugin after porting it for Deno. It's what pm2 uses. It would be a simpler, lightweight, standalone alternative to Node Exporter that could eventually be builtin to Pup's core.

**Comment by Hexagon** [comment]
Sounds great!!

It's ok to make changes to the plugin interface (and internals) if needed 👍
